File: 3/OOP_LR3_Galimov/Discount_product_Galimov.cpp
#include "pch.h"
#include "Discount_product_Galimov.h"

IMPLEMENT_SERIAL(Discount_product_Galimov, Product_Galimov, VERSIONABLE_SCHEMA | 0)

void Discount_product_Galimov::InputProduct()
{
	Product_Galimov::InputProduct();
	cout << "Input discount: ";
	cin >> discount;
	cout << "Input quantity: ";
	cin >> quantity;
}

void Discount_product_Galimov::OutputProduct()
{
	Product_Galimov::OutputProduct();
	cout << discount << "\n"
		<< quantity << "\n";
}

void Discount_product_Galimov::Serialize(CArchive& ar)
{
	Product_Galimov::Serialize(ar);
	if (ar.IsStoring())
	{
		ar << discount;
		ar << quantity;
	}
	else
	{
		ar >> discount;
		ar >> quantity;
	}
}

File: 5/OOP_LR4_Galimov/Discount_Product_Galimov.cs
using System;

namespace OOP_LR4_Galimov
{
    [Serializable]
    public class DiscountProduct_Galimov : Product_Galimov
    {
        private int discount;
        private int quantity;

        public DiscountProduct_Galimov() { }

        public override void InputProduct()
        {
            base.InputProduct();
            Console.Write("Input discount: ");
            discount = int.Parse(Console.ReadLine());
            Console.Write("Input quantity: ");
            quantity = int.Parse(Console.ReadLine());
        }

        public override void OutputProduct()
        {
            base.OutputProduct();
            Console.WriteLine($"{discount}\n{quantity}");
        }
    }
}
